### PR TITLE
Drop exim module

### LINKED ISSUE #1692: Replace exim Puppet module (or exim)
We currently have our own exim puppet module that could break in the future with newer operating systems. Moving to a shared puppet module may have a longer and better support life.

### PR DESCRIPTION
**Pull request by ekohl**
This module was appled on the old redmine.theforeman.org box, but the new redmine01.conova.theforeman.org no longer uses exim which means it can be dropped.

The question is if we should have some (minimal) postfix config that just ensures it's installed and running. We did see that on Jenkins where that now has been done manually.

Fixes #1692

### CODE REVIEW

**Comment by ekohl** [comment]
Rebased now that CI should be green again.

**Comment by ekohl** [comment]
> The question is if we should have some (minimal) postfix config that just ensures it's installed and running. We did see that on Jenkins where that now has been done manually.

That would be https://github.com/theforeman/foreman-infra/pull/1846

**Comment by ekohl** [comment]
I noticed we have exim on web01.

**Comment by ekohl** [comment]
And web01 is gone

**Review by evgeni — APPROVED** [review]

File: puppet/Rakefile [hunk 1]
```diff
@@ -11,7 +11,6 @@ PuppetSyntax.exclude_paths = ["vendor/**/*"]
 require 'puppet-lint/tasks/puppet-lint'
 PuppetLint::RakeTask.new :lint do |config|
   config.pattern = PuppetSyntax.manifests_paths
-  config.ignore_paths = ['modules/exim/**/*.pp']
```
Comment (ekohl) [inline puppet/Rakefile:14]: Rebased and updated this line to fully get rid of it. Waiting for CI now to see if it's still happy.

**Pull request merged**